core/src/actions/session.ts

```diff
@@ -25,16 +27,41 @@ export class SessionCreate extends CLSAction {
     params,
   }: {
     connection: Connection;
-    params: { [key: string]: string };
+    params: { email: string; password?: string; requestId?: string };
   }) {
+    if (!params.password && !params.requestId) {
+      throw new Errors.AuthenticationError(
+        `password or an oAuth requestId is required`
+      );
+    }
+
     const teamMember = await TeamMember.findOne({
       where: { email: params.email.toLocaleLowerCase() },
     });
-    if (!teamMember)
+    if (!teamMember) {
       throw new Errors.AuthenticationError("team member not found");
+    }
 
-    const match = await teamMember.checkPassword(params.password);
-    if (!match) throw new Errors.AuthenticationError("password does not match");
+    if (params.password) {
+      const match = await teamMember.checkPassword(params.password);
+      if (!match)
+        throw new Errors.AuthenticationError("password does not match");
+    } else {
+      const oauthRequest = await OAuthRequest.findOne({
+        where: { id: params.requestId },
+      });
```

Comment:
If this matches we should also delete the `OAuthRequest` or set something on it so it can't be used again. At the moment the `requestId` can be reused (until it's swept)

Reply:
👍 Love it.  Thank you for reading all this code so carefully!  Updated in e2ad6c612